internal/store/logs.go

```diff
@@ -82,7 +85,26 @@ func (s *Store) Collect(groupID string, newLog *opsmngr.LogCollectionJob) (*opsm
 func (s *Store) DownloadLog(groupID, host, name string, out io.Writer, opts *atlas.DateRangetOptions) error {
 	switch s.service {
 	case config.CloudService, config.CloudGovService:
-		_, err := s.client.(*atlas.Client).Logs.Get(s.ctx, groupID, host, name, out, opts)
+		fileBaseName := strings.TrimSuffix(name, filepath.Ext(name))
+		result := s.clientv2.MonitoringAndLogsApi.GetHostLogs(s.ctx, groupID, host, fileBaseName)
+		if opts != nil {
+			if opts.StartDate != "" {
```

Comment:
If this comes as `nil` instead of an empty string, what will the code do?

Reply:
In this particular example we dealing with string datatype that cannot be nil. 
https://github.com/mongodb/go-client-mongodb-atlas/blob/c3435ec31149df5973ef01f992bb72d598615a39/mongodbatlas/logs.go#L41-L42

Reply:
The problem is that we should not use v1 objects itself and v2 will actually have pointer

Reply:
`opts.StartDate` is declared as string 
```
type DateRangetOptions struct {
    StartDate string `url:"startDate,omitempty"`
    EndDate   string `url:"endDate,omitempty"`
}
```
ParseInt only accepts string value in the argument.